src/ray/core_worker/core_worker.cc

```diff
@@ -524,25 +523,34 @@ CoreWorker::CoreWorker(
   // NOTE: This also marks the worker as available in Raylet. We do this at the very end
   // in case there is a problem during construction.
   ConnectToRayletInternal();
+}
+
+CoreWorker::~CoreWorker() {
+  WaitForShutdownComplete();
+  RAY_LOG(INFO) << "Core worker is destructed";
+}
```

Comment:
if `Shutdown` were never called, this will return immediately. is that always safe to do? or could there be anything dangling that needs to be terminated in every case (like async callbacks)?